### PR TITLE
Add support for utilizing NOX capabilities as URL parameters during account creation

### PR DESCRIPTION
**Pull request by mordeth**
#### Changes proposed in this Pull Request
This PR manages the capabilities passed from the WooCommerce Payments NOX screen to the onboarding page and account creation process.

#### Testing instructions

> [!IMPORTANT]  
> Note: To test this PR, check out [this PR](https://github.com/woocommerce/woocommerce/pull/53688) on the WooCommerce core repository.

Case 1:
1. Navigate to wp-admin → Settings → Payments
2. From the country selector, choose United States.
3. Enable the WooPayments gateway.
4. Verify that the Recommended Payment Methods screen is displayed.
5. Confirm that the toggle buttons for payment methods are functional.
6. Click the Continue button and ensure you are redirected to the onboarding page and a test-drive account is created with the selected capabilities.

Case 2:
1. Navigate to wp-admin → Settings → Payments or directly visit: http://your.url/wp-admin/admin.php?page=wc-settings&tab=checkout.
2. Reset your account.
3. From the country selector, choose Bahamas.
4. Click Complete Setup.
5. Click the Continue button and ensure you are redirected to the onboarding page and a test-drive account is created with the all capabilities.

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9947` or branch name `update/pass-capabilities-to-onboarding-as-get-params` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/pass-capabilities-to-onboarding-as-get-params) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6cd46354f9c3e648417ecc577ee4ae54fdd2a0eb
- Build time: 2024-12-15 16:21:07 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-account.php [hunk 1]
```diff
@@ -1982,6 +1985,7 @@ private function init_stripe_onboarding( string $setup_mode, string $wcpay_conne
 		}
 
 		$self_assessment_data = isset( $_GET['self_assessment'] ) ? wc_clean( wp_unslash( $_GET['self_assessment'] ) ) : [];
+		$capabilities         = isset( $_GET['capabilities'] ) ? json_decode( wc_clean( wp_unslash( $_GET['capabilities'] ) ), true ) : [];
```
Comment (vladolaru) [inline includes/class-wc-payments-account.php]: See that json_decode will return `null` if it fails to decode. You should check for that and put in an empty array.

Comment (mordeth) [reply includes/class-wc-payments-account.php]: Good catch, fixed it.

**Review by oaratovskyi — APPROVED** [review]
Worked well! Code looks good.

**Pull request merged**